test/tests/functional/pbs_qrun.py

```diff
@@ -110,15 +109,16 @@ def test_qrun_hangs(self):
             j = Job(TEST_USER)
             a = {'Resource_List.walltime': walltime}
             j.set_attributes(a)
-            jid = self.server.submit(j)
-            joblist.append(jid)
+            if walltime == 500:
```

Comment:
you don't need this condition, you can just always say jid = self.server.submit(j) and then check for jid later, it will be job id of the last job submitted

Reply:
It will be an extra task to access and save a variable value each time (500 times) that we don't need.
Doing a conditional check before save the value should be more effective.
Let me know your suggestion.